src/auth/auth.service.ts

```diff
@@ -15,15 +15,17 @@ export class AuthService {
     async login(credentials: AuthCredentialsDTO): Promise<string> {
         const user = await this.validateUser(credentials);
         if (!user) {
-            throw new UnauthorizedException();
+            throw new UnauthorizedException(
+                'Username or Password is incorrect',
+            );
         }
         return this.sign(user);
     }
 
     async register(userDTO: User): Promise<string> {
         const user = await this.userService.create(userDTO);
         if (!user) {
-            throw new UnauthorizedException();
+            throw new UnauthorizedException('Username already exists');
         }
```

Comment:
This exception can be thrown by many error, not just a duplicate username. 
e.g. Request body violate database schema such as an empty `firstName` field.

Reply:
Have u tried to fill empty firstName? Are u sure it would throw an error here not the other part? For the record, I'm not sure 555

Reply:
But I think this is inaccurate error, because `!user` means user was not created not username already exists (In that case it would automatically throw error from database).